src/containers/Auth/Signup.tsx

```diff
@@ -71,45 +53,28 @@ function Signup() {
     options: {
       url: `/sign-up`,
       method: 'post',
-      data: { email, password },
+      data: { email: watch('email'), password: watch('password') },
     },
   });
 
-  const onSignup = async (e: SyntheticEvent) => {
-    e.preventDefault();
-    const newEmailErrorMessage = validateEmail(email);
-    const newPasswordErrorMessage = validatePassword(password, router.pathname);
-    const newPasswordCheckErrorMessage = validatePasswordCheck(
-      passwordCheck,
-      password,
-    );
-    if (
-      newEmailErrorMessage ||
-      newPasswordErrorMessage ||
-      newPasswordCheckErrorMessage
-    ) {
-      setEmailErrorMessage(newEmailErrorMessage);
-      setPasswordErrorMessage(newPasswordErrorMessage);
-      setPasswordCheckErrorMessage(newPasswordCheckErrorMessage);
-    }
-
-    if (!newEmailErrorMessage) {
-      const { error: duplicateEmailError } = await checkEmail();
-      if (duplicateEmailError) {
-        setEmailErrorMessage(ERROR_MESSAGES.email.unavailableEmail);
-        return;
-      }
+  const checkPasswordMatch = (value: string) => {
+    if (value !== watch('password')) {
+      return ERROR_MESSAGES.passwordCheck.invalidInput;
     }
+    return true;
+  };
 
-    if (
-      newEmailErrorMessage ||
-      newPasswordErrorMessage ||
-      newPasswordCheckErrorMessage
-    )
+  const onSignup = async () => {
```

Comment:
현재 formData를 통신 요청 시 전달하기 위해 watch를 사용해 필요한 값들을 `상태`로 관리하고 있습니다.
저번 멘토링에서 설명드린 것처럼 react-hook-form은 기존에 상태를 이용해 form의 값을 관리하는 `제어 컴포넌트` 대신에 `비제어 컴포넌트`로 관리할 수 있다는 것이 가장 큰 장점입니다.

가능하면 제가 아래 코드에 작성해 둔 것처럼 onSubmit의 인자로 받는 것이 가장 react-hook-form 다운 방식입니다.
요구사항에 따라서는 watch가 반드시 필요한 경우도 있으나 되도록이면 비제어 컴포넌트로 관리하는 방식을 추천드립니다.

```suggestion
  const onSignup = async (formData: FormValues) => {
    const { email, password, passwordCheck } = formData;
```

Reply:
다만, 이 부분을 변경하게 되면 이 코드에서 많은 부분이 바뀌어야 하므로 시간 여유가 되시는 경우에 반영하시면 좋을 것 같아요!

Reply:
예를 들면 이메일, 비밀번호 등에 대한 validation이 onSignup 내에서 이루어져야 한다는 변경 사항이 필요해 보이네요.

Reply:
대신 `비제어 컴포넌트`의 경우 코드의 복잡도가 감소하고, 성능적인 이점이 있다는 장점이 있으니 꼭 다른 구현 시에서라도 활용하시는 것을 추천드립니다! 🤗🤗